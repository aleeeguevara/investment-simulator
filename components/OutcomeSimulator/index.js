// 3rd parties
import { PropTypes } from 'prop-types';
// styles
import {
  Container, Cards, Card, Outcome,
} from './styles';

const OutcomeSimulator = function OutcomeSection({ simulationData, simulation }) {
  return (
    <Container simulation={simulation}>
      <h2>Resultado da Simulação</h2>
      <Cards>
        <Card>
          <h3>Valor Final Bruto</h3>
          <Outcome>{simulationData?.valorFinalBruto}</Outcome>
        </Card>

        <Card>
          <h3>Aliquota do IR</h3>
          <Outcome>{simulationData?.aliquotaIR}</Outcome>
        </Card>

        <Card>
          <h3>Valor Pago em IR</h3>
          <Outcome>{simulationData?.valorPagoIR}</Outcome>
        </Card>

        <Card>
          <h3>Valor Final Líquido</h3>
          <Outcome green>{simulationData?.valorFinalLiquido}</Outcome>
        </Card>

        <Card>
          <h3>Valor Total Investido</h3>
          <Outcome>{simulationData?.valorTotalInvestido}</Outcome>
        </Card>

        <Card>
          <h3>Ganho Liquido</h3>
          <Outcome green>{simulationData?.ganhoLiquido}</Outcome>
        </Card>

      </Cards>
    </Container>
  );
};
OutcomeSimulator.propTypes = {
  simulation: PropTypes.bool.isRequired,
  simulationData: PropTypes.shape({
    valorFinalBruto: PropTypes.number.isRequired,
    aliquotaIR: PropTypes.number.isRequired,
    valorPagoIR: PropTypes.number.isRequired,
    valorFinalLiquido: PropTypes.number.isRequired,
    valorTotalInvestido: PropTypes.number.isRequired,
    ganhoLiquido: PropTypes.number.isRequired,
  }).isRequired,
};
export default OutcomeSimulator;
