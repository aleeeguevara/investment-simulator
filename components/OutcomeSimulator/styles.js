import styled from 'styled-components';

export const Container = styled.section`
  display: ${({ simulation }) => (simulation ? 'flex' : 'none')};
  flex-direction: column;
  margin: 0;

  @media(min-width: 900px){
    margin: 0;
  }

`;
export const Cards = styled.div`

  @media (min-width: 900px){
    display: flex;
    flex-wrap: wrap;
    margin-top: 1rem;
    width: 400px;
  }
  @media (min-width: 1100px){
    width: 600px;
  }
`;
export const Card = styled.div`
  background: #f4f4f4;
  box-shadow:  0 0 3px #626262;
  width: 150px;
  height: 50px;
  margin: 1rem auto;
  padding: 5px;

  @media (min-width: 900px){
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    margin: 13px;
  }

  h3{
    font-weight: 700;
    font-size: 12px;
    margin: 0;
    text-align: center;
  }
`;
export const Outcome = styled.p`
  color: ${({ green }) => (green ? '#3fa130' : 'inherit')};
  font-size: 12px;
  margin: 0;
  text-align: center;
`;
