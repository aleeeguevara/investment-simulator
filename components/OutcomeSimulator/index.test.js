// 3rd parties
import { render, act } from '@testing-library/react';
import OutcomeSimulator from '.';

describe('Component of outcome from simulation', () => {
  it('The components snapshot should remain always the same', () => {
    act(() => {
      const { container } = render(<OutcomeSimulator
        true
        valorFinalBruto="2048.09"
        aliquotaIR="0"
        valorPagoIR="0"
        valorFinalLiquido="2048.09"
        valorTotalInvestido="1000"
        ganhoLiquido="1048.08"

      />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
