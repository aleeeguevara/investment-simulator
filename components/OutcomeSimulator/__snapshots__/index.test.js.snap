// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component of outcome from simulation The components snapshot should remain always the same 1`] = `
<section
  class="sc-bdvvtL dIUOie"
>
  <h2>
    Resultado da Simulação
  </h2>
  <div
    class="sc-gsDKAQ fdQESR"
  >
    <div
      class="sc-dkPtRN bVjIJK"
    >
      <h3>
        Valor Final Bruto
      </h3>
      <p
        class="sc-hKwDye enIhPe"
      />
    </div>
    <div
      class="sc-dkPtRN bVjIJK"
    >
      <h3>
        Aliquota do IR
      </h3>
      <p
        class="sc-hKwDye enIhPe"
      />
    </div>
    <div
      class="sc-dkPtRN bVjIJK"
    >
      <h3>
        Valor Pago em IR
      </h3>
      <p
        class="sc-hKwDye enIhPe"
      />
    </div>
    <div
      class="sc-dkPtRN bVjIJK"
    >
      <h3>
        Valor Final Líquido
      </h3>
      <p
        class="sc-hKwDye jqiquy"
      />
    </div>
    <div
      class="sc-dkPtRN bVjIJK"
    >
      <h3>
        Valor Total Investido
      </h3>
      <p
        class="sc-hKwDye enIhPe"
      />
    </div>
    <div
      class="sc-dkPtRN bVjIJK"
    >
      <h3>
        Ganho Liquido
      </h3>
      <p
        class="sc-hKwDye jqiquy"
      />
    </div>
  </div>
</section>
`;
